crates/store/src/db/sql/mod.rs

```diff
@@ -1002,6 +1029,91 @@ pub fn select_note_inclusion_proofs(
     Ok(result)
 }
 
+/// Returns a paginated batch of network notes that have not yet been consumed.
+///
+/// Pagination is controlled with the `offset` and `limit` values. If less than `limit` values are
+/// returned, then this is the final page.
+///
+/// # Returns
+///
+/// A batch of unconsumed network notes with maximum length of `limit`, from the given `offset`.
+#[expect(dead_code, reason = "gRPC method is not yet implemented")]
+pub fn unconsumed_network_notes(
+    transaction: &mut Transaction,
+    offset: usize,
+    limit: NonZeroUsize,
+) -> Result<Vec<NoteRecord>> {
+    assert_eq!(
+        NoteExecutionMode::Network as u8,
+        0,
+        "Hardcoded execution value must match query"
+    );
+
+    let mut stmt = transaction.prepare_cached(
+        "
+        SELECT
+            block_num,
+            batch_index,
+            note_index,
+            note_id,
+            note_type,
+            sender,
+            tag,
+            aux,
+            execution_hint,
+            merkle_path,
+            details,
+            nullifier
+        FROM
+            notes
+        WHERE
+            execution_mode = 0 AND consumed = FALSE
+        OFFSET ?0
+        LIMIT ?1
+        ",
+    )?;
+
+    let mut rows = stmt.query(params![offset, limit])?;
+
+    let mut notes = Vec::with_capacity(limit.into());
+    while let Some(row) = rows.next()? {
+        let block_num = read_block_number(row, 0)?;
+        let note_index = BlockNoteIndex::new(row.get(1)?, row.get(2)?)?;
+        let note_id = row.get_ref(3)?.as_blob()?;
+        let note_id = RpoDigest::read_from_bytes(note_id)?;
+        let note_type = row.get::<_, u8>(4)?.try_into()?;
+        let sender = AccountId::read_from_bytes(row.get_ref(5)?.as_blob()?)?;
+        let tag: u32 = row.get(6)?;
+        let aux: u64 = row.get(7)?;
+        let aux = aux.try_into().map_err(DatabaseError::InvalidFelt)?;
+        let execution_hint = column_value_as_u64(row, 8)?;
+        let merkle_path_data = row.get_ref(9)?.as_blob()?;
+        let merkle_path = MerklePath::read_from_bytes(merkle_path_data)?;
+        let details_data = row.get_ref(10)?.as_blob_or_null()?;
+        let details = details_data.map(<Vec<u8>>::read_from_bytes).transpose()?;
+        let nullifier = row
+            .get_ref(11)?
+            .as_blob_or_null()?
+            .map(Nullifier::read_from_bytes)
+            .transpose()?;
+
+        let metadata =
+            NoteMetadata::new(sender, note_type, tag.into(), execution_hint.try_into()?, aux)?;
```

Comment:
This code seems to be identical to the code on lines 758 - 779 and 943 - 964 above. Should we create a helper function which encapsulates it? Or maybe have something like `NoteRecord::from_row()` function?

Reply:
The concern I have is that it depends on the sql query ordering. So while they look the same now, thats just a consequence of them having the same field ordering in the string.

Reply:
Ah - I see. Maybe we could make the order of fields explicit too? - e.g. something like:
```rust
impl NoteRecord {
    pub const SELECT_FIELDS: &str = "
        block_num,
        batch_index,
        note_index,
        note_id,
        note_type,
        sender,
        tag,
        aux,
        execution_hint,
        merkle_path,
        details,
        nullifier";
}
```

Reply:
I'll try this out, but this does mean the sql str will allocate because you need to do

```rust
format!("SELECT {} FROM notes ...", NoteRecord::SELECT_FIELDS)
```

I think there *may* be some work-arounds with macros from other crates. Though probably this won't show up ito performance anyway.